src/plugins/data_source/server/saved_objects/data_source_saved_objects_client_wrapper.ts

```diff
@@ -116,100 +114,139 @@ export class CredentialSavedObjectsClientWrapper {
 
     return {
       ...wrapperOptions.client,
-      create: createWithCredentialMaterialsEncryption,
-      bulkCreate: bulkCreateWithCredentialMaterialsEncryption,
+      create: createWithCredentialsEncryption,
+      bulkCreate: bulkCreateWithCredentialsEncryption,
       checkConflicts: wrapperOptions.client.checkConflicts,
       delete: wrapperOptions.client.delete,
       find: wrapperOptions.client.find,
       bulkGet: wrapperOptions.client.bulkGet,
       get: wrapperOptions.client.get,
-      update: updateWithCredentialMaterialsEncryption,
-      bulkUpdate: bulkUpdateWithCredentialMaterialsEncryption,
+      update: updateWithCredentialsEncryption,
+      bulkUpdate: bulkUpdateWithCredentialsEncryption,
       errors: wrapperOptions.client.errors,
       addToNamespaces: wrapperOptions.client.addToNamespaces,
       deleteFromNamespaces: wrapperOptions.client.deleteFromNamespaces,
     };
   };
 
+  private isValidUrl(endpoint: string) {
```

Comment:
I remembered we talked about this may not be enough, and decide to use regular expression? @mpabba3003

Reply:
I haven't seen such logic in DSM, so let's make it a separate task.